interview_cake/buy_stocks/jessicas_solution/making_money.py

```diff
@@ -0,0 +1,44 @@
+def max_profit(stock_prices):
+    """
+    what is the max profit possible between two prices
+    of stock assuming you must buy before you can sell
+    in O(n)^2 time
+    """
+    max_profit = 0
+
+    for index, current_price in enumerate(stock_prices):
+        current_index = index
+        max_index = len(stock_prices)
+
+        for i in range(current_index+1, max_index):
+            selling_profit = stock_prices[i] - current_price
+
+            if selling_profit > max_profit:
+                max_profit = selling_profit
+
+    return max_profit
+
+def max_profit_faster(stock_prices):
+    """
+    make money faster in O(n) time
+    buy low then sell high later for max profits
+    """
+    if len(stock_prices) is 0:
+        return 0
+
+    open_price = stock_prices[0]
+    low_price = open_price
+    high_price = open_price
```

Comment:
`high_price` can be removed.  It's set during the loop (on line 40), but not used to determine the output.  `high_price` is used to set `max_profit`, it can still be removed (see other comment).

If `high_price` is removed, then `open_price` will only be used once and `open_price` can be removed by setting `low_price = stock_prices[0]`